### ISSUE TITLE
Curl against the website and get a formatted regex explainer

### ISSUE THREAD

**Issue by beyarkay** [gskinner/regexr#453]
## Feature

Hi! I love the website, use it multiple times a week to check my regex. Often my workflow is that I'll write regex, paste it into the website, and check some test cases or check the explainer to see that I haven't left anything unescaped or similar.

It would be _amazing_ if I could curl against the website and get back a formatted explainer of the regex, similar to how [cheat.sh](http://cheat.sh/) will explain bash commands via `curl http://cheat.sh/pwd`. Something like:

```
$ curl regexr.com/explain/[a-z]+
[a-z]+
     ^- one or more times
 ^^^^- any letter from a to z
```
This means that from the commandline I can directly paste explainers of complicated regex as comments into my code which would be super useful.

There would be some trickery to escaping regex characters in the URL, but I'm happy to help out if this feature is put on the roadmap.

### For reference
This is how cheat.sh works:
```
$ curl cheat.sh/pwd
 cheat:pwd
# Show the absolute path of your current working directory:
pwd

 tldr:pwd
# pwd
# Print name of current/working directory.

# Print the current directory:
pwd

# Print the current directory, and resolve all symlinks (i.e. show the "physical" path):
pwd -P
```
It also colour codes the various parts, but that's more of a nice-to-have
